airflow/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -956,11 +956,10 @@ def _list_request(
                 next_token = response["NextToken"]
 
     def find_processing_job_by_name(self, processing_job_name: str) -> bool:
-        """
-        Query processing job existence by describe_processing_job, rather than iterate all the jobs.
-        """
         try:
             res = self.get_conn().describe_processing_job(ProcessingJobName=processing_job_name)
             return True
         except ClientError as e:
-            return False
+            if e.response['Error']['Code'] == 'ValidationException':
+                return False
+            raise e
```

Comment:
```suggestion
            raise
```

This preserves the original traceback, which is likely more useful for the user.

Reply:
updated